pyaedt/filtersolutions_core/dll_interface.py

```diff
@@ -0,0 +1,192 @@
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import ctypes
+from enum import Enum
+import os
+import threading
+import time
+from typing import Callable
+
+from pyaedt.misc.misc import current_version
+from pyaedt.misc.misc import installed_versions
+
+
+class DllInterface:
+    """Interfaces with the FilterSolutions C++ API DLL."""
+
+    def __init__(self, show_gui=False, version=None):
+        self.version = version
+        self._init_dll_path()
+        self._init_dll(show_gui)
+
+    def restore_defaults(self):
+        """Restore the state of the API, including all options and values, to the initial startup state."""
+        status = self._dll.startApplication(self.show_gui)
+        self.raise_error(status)
+
+    def _init_dll_path(self):
+        """Set DLL path and print the status of the DLL access to the screen."""
+        self_current_version = current_version()
+        if current_version() == "":
+            raise Exception("AEDT is not installed on your system. Install AEDT version 2025 R1 or higher.")
+        if self.version is None:
+            self.version = self_current_version
+        if float(self.version[0:6]) < 2025:
+            raise ValueError("PyAEDT supports AEDT version 2025 R1 and later. Recommended version is 2025 R1 or later.")
+        if not (self.version in installed_versions()) and not (self.version + "CL" in installed_versions()):
+            raise ValueError("Specified version {} is not installed on your system".format(self.version[0:6]))
+        if self.version != self_current_version:
+            self.dll_path = os.path.join(installed_versions()[self.version], "nuhertz/FilterSolutionsAPI.dll")
+        else:
+            self.dll_path = os.path.join(installed_versions()[current_version()], "nuhertz/FilterSolutionsAPI.dll")
+            # pragma: no cover
+        print("DLL Path:", self.dll_path)
+        if not os.path.isfile(self.dll_path):
+            raise RuntimeError(f"The 'FilterSolution' API DLL was not found at {self.dll_path}.")  # pragma: no cover
```

Comment:
```suggestion
    def __init__(self, show_gui=False, version=None):
        self._init_dll_path(version)
        self._init_dll(show_gui)

    def restore_defaults(self):
        """Restore the state of the API, including all options and values, to the initial startup state."""
        status = self._dll.startApplication(self.show_gui)
        self.raise_error(status)

    def _init_dll_path(self, version):
        """Set DLL path and print the status of the DLL access to the screen."""
        latest_version = current_version()
        if latest_version == "":
            raise Exception("AEDT is not installed on your system. Install AEDT version 2025 R1 or higher.")
        if version is None:
            version = latest_version
        if float(version[0:6]) < 2025:
            raise ValueError("PyAEDT supports AEDT version 2025 R1 and later. Recommended version is 2025 R1 or later.")
        if not (version in installed_versions()) and not (version + "CL" in installed_versions()):
            raise ValueError("Specified version {} is not installed on your system".format(version[0:6]))
        self.dll_path = os.path.join(installed_versions()[self.version], "nuhertz/FilterSolutionsAPI.dll")
        print("DLL Path:", self.dll_path)
        if not os.path.isfile(self.dll_path):
            raise RuntimeError(f"The 'FilterSolution' API DLL was not found at {self.dll_path}.")  # pragma: no cover
```

Reply:
@ramin4667 I refactored _init_dll_path a bit to accomplish a few things:

1. self.version isn't used anywhere else in the class so I made it an argument to _init_dll_path to make the relationship clearer.
2. I renamed self_current_version to latest_version since the meaning of self wasn't clear and latest seemed clearer. "Current" could mean currently selected, when in fact this is the newest or latest installed version.
3. I removed the if/else because it looks like it uses the value of version either way (if self.version != current version it was using self.version. if they were equal it was using current_version which is also self.version).